app/models/watchlist_instance.rb

```diff
@@ -506,6 +513,39 @@ def self.add_new_instance_for_cud_grace_day_usage(course, condition_id,
                           violation_info: violation_info)
   end
 
+  def self.add_new_instance_for_cud_extension_requests(course, condition_id, cud, extension_count)
+    asmts = Assessment.where(course_id: course.id).pluck(:id)
+    violation_info = {}
+    num_of_extensions = 0
+
+    asmts.each do |asmt|
+      extensions = Extension.where(course_user_datum_id: cud.id, assessment_id: asmt)
+      extensions.each do
+        num_of_extensions += 1
+      end
+    end
+
+    if num_of_extensions >= extension_count
+      asmts.each do |asmt|
+        extensions = Extension.where(course_user_datum_id: cud.id, assessment_id: asmt)
+        aud = AssessmentUserDatum.find_by(course_user_datum_id: cud.id, assessment_id: asmt)
+        indiv_extensions_count = 0
+        extensions.each do
```

Comment:
similarly here, an `.count` should be used https://api.rubyonrails.org/classes/ActiveRecord/Calculations.html#method-i-count

Reply:
Fixed.